File: laramie-tesoreria-webapi/src/domain/entities/movimiento-caja.ts
export default class MovimientoCaja {

    id: number;
	idCaja: number;
	idCajaAsignacion: number;
	idTipoMovimientoCaja: number;
	importeCobro: number;
	fechaCobro: Date;
	observacion: string;

	constructor(
        id: number = 0,
		idCaja: number = 0,
		idCajaAsignacion: number = 0,
		idTipoMovimientoCaja: number = 0,
		importeCobro: number = 0,
		fechaCobro: Date = null,
		observacion: string = ""
	)
	{
        this.id = id;
		this.idCaja = idCaja;
		this.idCajaAsignacion = idCajaAsignacion;
		this.idTipoMovimientoCaja = idTipoMovimientoCaja;
		this.importeCobro = importeCobro;
		this.fechaCobro = fechaCobro;
		this.observacion = observacion;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.idCaja = row.idCaja ?? 0;
		this.idCajaAsignacion = row.idCajaAsignacion ?? 0;
		this.idTipoMovimientoCaja = row.idTipoMovimientoCaja ?? 0;
		this.importeCobro = row.importeCobro ?? 0;
		this.fechaCobro = row.fechaCobro ?? null;
		this.observacion = row.observacion ?? "";
	}

}

File: modules/tesoreria/models/movimiento_caja.py
from sqlalchemy import Column, BigInteger, String, Numeric, DateTime

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class MovimientoCaja(Base):
    __tablename__ = "tesoreria_movimientos_caja"

    id = Column(BigInteger, primary_key=True)
    id_caja = Column(BigInteger, nullable=False)
    id_caja_asignacion = Column(BigInteger, nullable=False)
    id_tipo_movimiento_caja = Column(BigInteger, nullable=False)
    importe_cobro = Column(Numeric(18, 2), nullable=False)
    fecha_cobro = Column(DateTime, nullable=False)
    observacion = Column(String(500), nullable=True)
